platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/componentframework/TaskProcessors.java

```diff
@@ -0,0 +1,211 @@
+                            b.getDefinition().getName()));
+                })
+                .orElseThrow(() -> new IllegalArgumentException(String.format(
+                        "Implementing class %s does not implement any TaskProcessor definitions",
+                        implementation.getClass().getName())))
+                .setImplementation(implementation);
+    }
+
+    /**
+     * Same as {@link #addImplementation(TaskProcessor)} but has a second parameter to explicitly specify the type. This
+     * is useful for passing in lambdas because the type cannot be inferred.
+     */
+    public <T extends TaskProcessor> void addImplementation(
+            @NonNull final T component, @NonNull final Class<T> componentClass) {
+        throwIfStarted();
+        Objects.requireNonNull(component);
+        Objects.requireNonNull(componentClass);
+        parts.get(componentClass).setImplementation(component);
+    }
+
+    /**
+     * Returns the submitter for the given task processor class. The submitter is used to submit tasks to the task
+     * processor to handle asynchronously.
+     *
+     * @param processorClass
+     * 		the class of the task processor to get the submitter for
+     * @return the submitter for the given task processor class
+     */
+    @SuppressWarnings("unchecked")
+    public <T extends TaskProcessor> @NonNull T getSubmitter(@NonNull final Class<T> processorClass) {
+        Objects.requireNonNull(processorClass);
+        return (T) parts.get(processorClass).getSubmitter();
+    }
+
+    /**
+     * Start the framework. This will start all the task processors in their own threads.
+     */
+    @SuppressWarnings("unchecked")
+    public void start() {
+        throwIfStarted();
+        parts.values().stream().filter(p -> p.getImplementation() == null).forEach(p -> {
+            throw new IllegalStateException(String.format(
+                    "TaskProcessor %s has no implementation set",
+                    p.getDefinition().getName()));
+        });
+
+        for (final ProcessorParts processorParts : parts.values()) {
+            final Map<Class<?>, InterruptableConsumer<?>> processingMethods =
+                    processorParts.getImplementation().getProcessingMethods();
+            final InterruptableConsumer<?> handler;
+            if (processingMethods.size() == 1) {
+                handler = processingMethods.values().iterator().next();
+            } else {
+                handler = new MultiHandler(processingMethods)::handle;
+            }
+
+            final QueueThread<Object> queueThread = defaultConfiguration
+                    .copy()
+                    .setThreadName(processorParts.getConfig().name())
+                    .setQueue(processorParts.getQueue())
+                    .setHandler((InterruptableConsumer<Object>) handler)
+                    .build(true);
+            processorParts.setQueueThread(queueThread);
+        }
```

Comment:
How does this code behave if you don't provide it with an implementation for a particular task processor type?